nb-api/src/main/java/io/nosqlbench/adapters/api/util/TagFilter.java

```diff
@@ -107,6 +105,12 @@ public <T extends Tagged> List<T> filter(List<T> tagged) {
             .collect(Collectors.toList());
     }
 
+    public <T extends NBLabeledElement> List<T> filterLabeled(Collection<T> labeled) {
+        return labeled.stream()
+            .filter(l -> this.matches(l.getLabels().asMap()).matched())
```

Comment:
It seems a little confusing to have both an instance variable and a method named filter (in addition to the streaming filter method)